### DISCUSSION TITLE
Full native netbeans Laravel support

### DISCUSSION THREAD

**Discussion by parallels999** [apache/netbeans#7231] [Ideas]
It would be great if it had full laravel support

For blade there is a plugin (_only blade files_): https://github.com/haidubogdan/netbeans-php-blade-plugin
It has great support but it's still buggy.

`CTRL + Click` 
![use](https://github.com/apache/netbeans/assets/109294935/a4bbb4e4-05a0-47ca-a95f-d910b74ba930)
autocomplete paths
![screenshot-view](https://github.com/apache/netbeans/assets/109294935/0f8b74c3-b85f-4fdb-866f-0a56e38accc6)
config paths, there is files with array, so they are array assoc paths [laravel/laravel/config](https://github.com/laravel/laravel/tree/11.x/config)
![screenshot-config](https://github.com/apache/netbeans/assets/109294935/e80457b8-e0f7-4ef7-91d0-9612e0fd61ae)
route paths, laravel has route files, [laravel/laravel/routes](https://github.com/laravel/laravel/tree/11.x/routes)
![screenshot-route](https://github.com/apache/netbeans/assets/109294935/b7456e82-dda3-42e2-a3cd-41265fe525f8)
Translations and translation parameters, [Laravel/laravel/lang/en](https://github.com/laravel/laravel/tree/9.x/lang/en)
![screenshot-trans](https://github.com/apache/netbeans/assets/109294935/a35b8101-88ce-4f52-bd78-3e3662654c9b)

Laravel also has a list of commands to create models, controllers and others, for example
```sh
php artisan make:model Flight
```

code templates for laravel could be added

I see tools for other PHP frameworks, Laravel is many years old but no tools have been added

![image](https://github.com/apache/netbeans/assets/109294935/b48cb973-424b-4553-b065-d03235eeeb9b)

https://github.com/nbphpcouncil/nb-laravel-plugin

**Comment by matthiasblaesing** [comment]
> It would be great if it had full laravel support

This will need people who are willing to do the work and integrate that into the existing PHP support. The license of the referenced new plugin will prevent it from ever being merged to Apache NetBeans as the Apache Foundation has a strict ALv2 policy and GPL v3 is only one-way compatible.

**Comment by matthiasblaesing** [comment]
@parallels999 is the discussion over?! My point is just, that it does not help to enumerate feature, it needs people to do the work.

**Reply by parallels999** [reply]
Yes, that would be the most difficult part, and since that hasn't existed after so long, I don't think it will happen now.

**For example:** I found this laravel ref

https://github.com/nbphpcouncil/nb-laravel-plugin
https://github.com/apache/netbeans/blob/9bc649bac0bdd83d51a71cce80759252dd213b00/php/php.composer/nbproject/project.xml#L258

I think that in this case it would be more feasible to move to another IDE that already has support

https://github.com/apache/netbeans/issues/5965

**Reply by haidubogdan** [reply]
> The license of the referenced new plugin will prevent it from ever being merged to Apache NetBeans as the Apache Foundation has a strict ALv2 policy and GPL v3 is only one-way compatible.

If it's regarding my plugin https://github.com/haidubogdan/netbeans-php-blade-plugin , I can adapt the license it's not really relevant.

For the moment my plugin is still in development mode, it's not only the existing bugs which concerns me ... but with performance impact. I don't want my plugin to affect the IDE performance with php projects.

---

> For example: I found this laravel ref
>
> https://github.com/nbphpcouncil/nb-laravel-plugin

@parallels999 An implementation similar to this plugin is much easier to do than the blade editor :) . 

The main issue with php framework extension is that foreach netbeans version I need to have a individual plugin, the php framework extension changes with each version. 
Yet, this should be manageable now with github action.

Depending on my available time, I can try o participate in the netbeans request feature to add at least a framework extension, starting with **netbeans 23**

Indeed it's all about time, skills and manpower.

**Reply by Chris2011** [reply]
I will also have a look into your nice plugin @haidubogdan. I can add the thing, that laravel should only be available, if the PHP project uses the framework laraval and enables it like for other frameworks too. Not soon, but I will have a look :)

**Reply by Chris2011** [reply]
Btw, thx for your great work.

**Reply by parallels999** [reply]
> https://github.com/haidubogdan/netbeans-php-blade-plugin

I'm using it on the blade files, it's pretty stable at the moment, I haven't had any problems.

**Reply by haidubogdan** [reply]
Cool, thank you :) .
I will try to wrap some unfinished flows, and next week I will prepare to deploy it to the maven repository to be accessible as a netbeans plugin. After that I can begin the process to included it in netbeans.

---

I'm preparing a laravel framework repo : https://github.com/haidubogdan/netbeans-php-laravel-framework .
I'm thinking of adding a "customized support" for php commands to use either docker or remote local vm's for Php scripts.

I will inform you when something is ready for testing.

**Reply by Chris2011** [reply]
If one day will come to implement this into the NetBeans core, we need a different license than GPL. Please use MIT or Apache 2 if you own both repos.

**Comment by haidubogdan** [comment]
As a update for this discussion.
I've opened issue #7531 . (Still triage). I hope to be treated soon.
I've forked **netbeans** repo to https://github.com/haidubogdan/netbeans/tree/7531 which will contain further development for the blade and laravel plugin.
